api/src/main/java/ca/bc/gov/educ/studentdatacollection/api/orchestrator/UpdateStudentDownstreamOrchestrator.java

```diff
@@ -58,22 +69,110 @@ public void updateStudent(final Event event, final SdcSagaEntity saga, final Upd
-        updateUsualNameFields(studentDataFromEventResponse, updateStudentSagaData);
-        studentDataFromEventResponse.setPostalCode(updateStudentSagaData.getPostalCode());
-        studentDataFromEventResponse.setHistoryActivityCode("REQMATCH");
-
-        final Event nextEvent = Event.builder().sagaId(saga.getSagaId())
-                .eventType(UPDATE_STUDENT)
-                .replyTo(this.getTopicToSubscribe())
-                .eventPayload(JsonUtil.getJsonStringFromObject(studentDataFromEventResponse))
-                .build();
-        this.postMessageToTopic(STUDENT_API_TOPIC.toString(), nextEvent);
-        log.debug("message sent to STUDENT_API_TOPIC for UPDATE_STUDENT Event.");
+        final List<SdcSchoolCollectionStudentEntity> otherStudentsWithSameAssignedID = sdcSchoolCollectionStudentRepository.findAllDuplicateStudentsByCollectionID(updateStudentSagaData.getCollectionID(), Collections.singletonList(updateStudentSagaData.getAssignedStudentID()));
+
+        if (updateStudentSagaData.getCollectionTypeCode().equalsIgnoreCase(CollectionTypeCodes.JULY.getTypeCode()) ||
+                otherStudentsWithSameAssignedID.isEmpty() ||
+                isStudentAttendingSchoolOfRecord(updateStudentSagaData, otherStudentsWithSameAssignedID)){
+
+            studentDataFromEventResponse.setUpdateUser(ApplicationProperties.STUDENT_DATA_COLLECTION_API);
+            studentDataFromEventResponse.setMincode(updateStudentSagaData.getMincode());
+            studentDataFromEventResponse.setLocalID(updateStudentSagaData.getLocalID());
+            studentDataFromEventResponse.setGradeCode(updateStudentSagaData.getGradeCode());
+            updateGradeYear(studentDataFromEventResponse, updateStudentSagaData);
+            updateUsualNameFields(studentDataFromEventResponse, updateStudentSagaData);
+            studentDataFromEventResponse.setPostalCode(updateStudentSagaData.getPostalCode());
+            studentDataFromEventResponse.setHistoryActivityCode("SLD");
+
+            final Event nextEvent = Event.builder().sagaId(saga.getSagaId())
+                    .eventType(UPDATE_STUDENT)
+                    .replyTo(this.getTopicToSubscribe())
+                    .eventPayload(JsonUtil.getJsonStringFromObject(studentDataFromEventResponse))
+                    .build();
+            this.postMessageToTopic(STUDENT_API_TOPIC.toString(), nextEvent);
+            log.debug("message sent to STUDENT_API_TOPIC for UPDATE_STUDENT Event.");
+
+        } else {
+            final Event nextEvent = Event.builder().sagaId(saga.getSagaId())
+                    .eventType(UPDATE_STUDENT)
+                    .eventOutcome(NO_STUDENT_UPDATE_NEEDED)
+                    .eventPayload(JsonUtil.getJsonStringFromObject(updateStudentSagaData))
+                    .build();
+            this.postMessageToTopic(this.getTopicToSubscribe(), nextEvent);
+            log.debug("message sent to UPDATE_STUDENT_DOWNSTREAM_SAGA_TOPIC for STUDENT_NOT_UPDATED Event.");
```

Comment:
Adjust the message for the event outcome of NO_STUDENT_UPDATE_NEEDED.

Reply:
yes - good catch, ty!